### PR TITLE
libbpf-tools: fix runqlen's error handling logic

### PR DESCRIPTION
**Pull request by ethercflow**
fix similar bugs  refer to https://github.com/iovisor/bcc/pull/3131

Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### CODE REVIEW

File: libbpf-tools/runqlen.c [hunk 1]
```diff
@@ -45,17 +45,19 @@ const char *argp_program_bug_address = "<bpf@vger.kernel.org>";
...
+"USAGE: runqlen [--help] [-C] [-O] [-T] [-f FREQUENCY] [interval] [count]\n"
 "\n"
 "EXAMPLES:\n"
 "    runqlen         # summarize run queue length as a histogram\n"
 "    runqlen 1 10    # print 1 second summaries, 10 times\n"
 "    runqlen -T 1    # 1s summaries and timestamps\n"
 "    runqlen -O      # report run queue occupancy\n"
-"    runqlen -C      # show each CPU separately\n";
+"    runqlen -C      # show each CPU separately\n"
+"    runqlen -f 199  # sample as 199hz\n";
```
Comment (anakryiko) [inline libbpf-tools/runqlen.c:56]: typo: sample at 199HZ

File: libbpf-tools/runqlen.c [hunk 2]
```diff
@@ -222,22 +235,22 @@ int main(int argc, char **argv)
...
+	if (nr_cpus < 0) {
+		printf("failed to get # of possible cpus: '%s'!\n",
+		       strerror(-nr_cpus));
 		return 1;
 	}
-
-	nr_cpus = libbpf_num_possible_cpus();
 	if (nr_cpus > MAX_CPU_NR) {
-		fprintf(stderr, "The number of cpu cores is too much, please "
+		fprintf(stderr, "the number of cpu cores is too much, please "
```
Comment (anakryiko) [inline libbpf-tools/runqlen.c:245]: probably should say "too big", not "too much". "Much" is used for uncountable entities.

File: libbpf-tools/runqlen.c [hunk 3]
```diff
@@ -222,22 +235,22 @@ int main(int argc, char **argv)
...
 		return 1;
 	}
-	links = calloc(nr_cpus, sizeof(*links));
-	if (!links) {
-		fprintf(stderr, "failed to alloc links\n");
-		goto cleanup;
+
+	obj = runqlen_bpf__open();
+	if (!obj) {
+		fprintf(stderr, "failed to open and/or load BPF object\n");
```
Comment (anakryiko) [inline libbpf-tools/runqlen.c:252]: just open, not yet load

File: libbpf-tools/runqlen.c (hunk 1 above)
Comment (ethercflow) [reply libbpf-tools/runqlen.c]: Fixed, thanks! Also fixed this in https://github.com/iovisor/bcc/pull/3131.

File: libbpf-tools/runqlen.c (hunk 2 above)
Comment (ethercflow) [reply libbpf-tools/runqlen.c]: Fixed, thanks!

File: libbpf-tools/runqlen.c (hunk 3 above)
Comment (ethercflow) [reply libbpf-tools/runqlen.c]: Fixed, thanks! I'll check other tools as well and fix them in next PR.

**Comment by yonghong-song** [comment]
[buildbot, test this please]

**Review by anakryiko — APPROVED** [review]

**Pull request merged**